### PR TITLE
Add Infraction Tests to the testing suite

### PR DESCRIPTION
**Pull request by D0rs4n**
The PR adds Infractions tests to the testing suite (Just like the title says), and it also introduces minor consistency enhancements in the testing suite.

### CODE REVIEW

**Review by ChrisLovering — APPROVED** [review]
Looks good to me, tests all pass locally, nice job!

For the readme, could you add some info on how to run these tests under docker?

IE 
- create a `.env` file with `DATABASE_URL=postgresql+asyncpg://codejam_management:codejam_management@localhost:7777/codejam_management` in it
- run `docker-compose up`
- run `pytest` locally

**Comment by D0rs4n** [comment]
Yeah, sure thing!

File: tests/routers/conftest.py [hunk 1]
```diff
@@ -58,4 +59,27 @@ async def created_codejam(
...
+        session: AsyncSession,
+        created_codejam: models.CodeJamResponse
+) -> models.InfractionResponse:
+    """Create a test Infraction via the API and yield it."""
+    # Select one of the test users, so that we can issue an infraction to that user
+    user_id = created_codejam.teams[0].users[0].user_id
+    jam_id = created_codejam.id
+    response = await client.post(
+        app.url_path_for("create_infraction"),
+        json={"user_id": user_id, "jam_id": jam_id, "reason": "To good to be true", "infraction_type": "warning"}
```
Comment (ToxicKidz) [inline tests/routers/conftest.py:77]:
```suggestion
        json={"user_id": user_id, "jam_id": jam_id, "reason": "Too good to be true", "infraction_type": "warning"}
```

**Review by ToxicKidz — APPROVED** [review]
Looks good, just one typo to fix. Thanks for the change!

File: tests/routers/conftest.py (hunk 1 above)
Comment (D0rs4n) [reply tests/routers/conftest.py]: Rookie mistake, thanks! 👍

**Pull request merged**